x/txreward/internal/types/blockrewardspool.go

```diff
@@ -0,0 +1,25 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type BlockRewardsPool struct {
```

Comment:
I really don't think that this type in necessary. We can simply use the `sdk.DecCoins`  (or even `sdkCoins`) type in order to represent the current pool funds amount making the whole code more DRY.